detox/test/e2e/28.visibility.test.js

```diff
@@ -13,9 +13,27 @@ describe('Visibility Cases', () => {
       .scroll(200, 'right');
   });
 
-  it('should be able to tap on the main ScrollView regardless of the on-screen keyboard', async () => {
+  it(':ios: should be able to tap on the main ScrollView in a blind way, on the center, when the on-screen keyboard is enabled', async () => {
     await element(by.id('screenScroll')).tap();
     await element(by.id('inputExample')).tap();
+
+    const { frame: { x, y, width, height } } = await element(by.id('screenScroll')).getAttributes();
+
+    await element(by.id('screenScroll')).tap({
+      x: Math.floor(x + 0.5 * width),
```

Comment:
The coordinates are in the element's coordinate space, so it doesn't make sense to add x and y.

Reply:
Fixed.

Reply:
I still have the error:

```
  ● Visibility Cases › :ios: should be able to tap on the main ScrollView in a blind way, on the center, when the on-screen keyboard is enabled

    Test Failed: View “<RCTScrollView: 0x7ff1cd536f60>” is not visible: view does not pass visibility threshold (75%)
    TIP: To print view hierarchy on failed actions/matches, use loglevel verbose and above.

      19 |     const { elementSafeBounds: bounds } = await element(by.id('screenScroll')).getAttributes();
      20 |
    > 21 |     await element(by.id('screenScroll')).tap({
         |                                          ^
      22 |       x: Math.floor(0.5 * bounds.width),
      23 |       y: Math.floor(0.5 * bounds.height),
      24 |     });

      at Object.it.only (test/e2e/28.visibility.test.js:21:42)
```

Reply:
🤷‍♂️ Is it visible?

Reply:
Yes, but the algorithm thinks it is not. Showing the problem on the example of the internal E2E tests.

| Human | Machine |
|:-----:|:-------:|
|<img width="360" alt="DETOX_VISIBILITY_RCTScrollView  0x7fcaf9dd0af0 _SCREEN" src="https://user-images.githubusercontent.com/1962469/94303840-25e24b80-ff77-11ea-8ff8-f8a2b28eb896.png">|<img width="360" alt="DETOX_VISIBILITY_RCTScrollView  0x7fcaf9dd0af0 _TEST" src="https://user-images.githubusercontent.com/1962469/94303833-22e75b00-ff77-11ea-8129-d983056a42e3.png">|

Reply:
Probably not enough percentage of visibility. Other than implementing a visibility policy setting, I don't see what else can be done here.

Reply:
Can we check absolute value of visible area? E.g., if it is about 40pxx40px area (or whatever Apple HIG say) AND 50% of area is visible, then tap can be executed... 🤔

Reply:
I don’t understand what that means.

Reply:
Also, keyboard has different proportions on different devices.

Reply:
How do you compare against 75% threshold? What is your formula to calculate the total area vs visible area?

Reply:
Count visible pixels, divide by total.

Reply:
When you count visible pixels, can you tell what the biggest visible area region is, and what its dimensions?